src/bscPlugin/transpile/BrsFilePreTranspileProcessor.ts

```diff
@@ -178,6 +181,138 @@ export class BrsFilePreTranspileProcessor {
         }
     }
 
+    private processNullCoalescingExpression(nullCoalescingExpression: NullCoalescingExpression, visitor: ReturnType<typeof createVisitor>, walkMode: WalkMode) {
+        // Check if this null coalescing expression has complex expressions that require scope protection
+        const consequentInfo = util.getExpressionInfo(nullCoalescingExpression.consequent, this.event.file);
+        const alternateInfo = util.getExpressionInfo(nullCoalescingExpression.alternate, this.event.file);
+
+        let hasComplexExpression = [
+            ...consequentInfo.expressions,
+            ...alternateInfo.expressions
+        ].find(e => isCallExpression(e) || isCallfuncExpression(e) || isDottedGetExpression(e) || isIndexedGetExpression(e));
+
+        // Only optimize if there are no complex expressions
+        if (hasComplexExpression) {
+            return;
+        }
+
+        function getOwnerAndKey(statement: Statement) {
+            const parent = statement.parent;
+            if (isBlock(parent) || isBody(parent)) {
+                let idx = parent.statements.indexOf(statement);
+                if (idx > -1) {
+                    return { owner: parent.statements, key: idx };
+                }
+            }
+        }
+
+        //if the null coalescing expression is part of a simple assignment to a local variable, rewrite it as an `IfStatement`
+        let parent = nullCoalescingExpression.findAncestor(x => !isGroupingExpression(x));
+        let operator: Token;
+        //operators like `+=` will cause the RHS to be a BinaryExpression due to how the parser handles this. let's do a little magic to detect this situation
```

Comment:
Let's exclude operators like `+=` for the scope of this work. Just focus on simple local variable assignment statements. 

Be sure to remove the logic below that was handling this. we should probably only have the `if(isassignmentstatement)` block.